azurerm_redis_cache/vars.tf

```diff
@@ -0,0 +1,97 @@
+variable "global_prefix" {
+  type = string
+  description = "Prefix used to define the resource name."
+}
+
+variable "environment" {
+  type          = string
+  description   = "The name of the environment"
+}
+
+variable "region" {
+  type = string
+  description = "The location of the resource group."
+}
+
+variable "name" {
+  type        = string
+  description = "The name of the Redis instance."
+}
+
+variable "resource_group_name" {
+  type = string
+}
+
+variable "capacity" {
+  type          = number
+  description   = "The size of the Redis cache to deploy"
+  default       = 1
+}
+
+variable "shard_count" {
+    type        = number
+    description = "The number of Shards to create on the Redis Cluster."
+    default     = null
+}
+
+variable "enable_non_ssl_port" {
+    type        = bool
+    description = "Enable the non-SSL port (6379) - disabled by default."
+    default     = false
+}
+
+variable "minimum_tls_version" {
+    type        = string
+    description = "The minimum TLS version."
+    default     = "1.0"
+}
+
+variable "subnet_id" {
+    type        = string 
+    description = "The Subnet within which the Redis Cache should be deployed."
+    default     = "" 
```

Comment:
This means that you can NOT pass a subnet_id and the default value of "" will be passed to the module, but if you pass "" as value to the module you will cause an error because subnet_id can't be empty.